src/main/java/edu/umich/its/cpm/BoxUtils.java

```diff
@@ -612,7 +612,23 @@ public static BoxAPIConnection getBoxAPIConnection(String userId, BoxAuthUserRep
 			// make connection
 			BoxAPIConnection api = new BoxAPIConnection(boxClientId,
 					boxClientSecret, boxAccessToken, boxRefreshToken);
-			
+			try
+			{
+				// a test call to check whether the API is valid
+				// the result of api.needsRefresh() is not useful
+				BoxUser.getCurrentUser(api);
+			}
+			catch (BoxAPIException ee)
+			{
+				if (ee.getResponseCode() == org.apache.http.HttpStatus.SC_UNAUTHORIZED)
+				{
+					// need to update access token
+					api.refresh();
```

Comment:
There is IllegalStateException thrown in that case. 

Added catch section for that exception, and the function will return null value.